crates/data_components/src/graphql/provider.rs

```diff
@@ -64,14 +84,26 @@ impl TableProvider for GraphQLTableProvider {
         filters: &[Expr],
         limit: Option<usize>,
     ) -> datafusion::error::Result<Arc<dyn ExecutionPlan>> {
-        let res = self
+        let mut res = self
             .client
-            .execute_paginated(Arc::clone(&self.schema), limit)
+            .execute_paginated(Arc::clone(&self.gql_schema), limit)
             .await
             .boxed()
             .map_err(DataFusionError::External)?;
 
-        let table = MemTable::try_new(Arc::clone(&self.schema), res)?;
+        if let Some(transform_fn) = &self.transform_fn {
```

Comment:
I don't know if it's needed (because it doesn't hurt performance that much), but i feel like it should be possible to only perform certain transform_fn if the projection contains certain keys. Probably unnecessary,

Reply:
That means we'll need to define list of fields, right?

Reply:
Merging as is for now, we'll check possible improvements separately